nx-X11/programs/Xserver/hw/nxagent/Events.c

```diff
@@ -426,7 +426,7 @@ void nxagentRemoteWindowsTree(Window window, int level)
   int i, j;
   unsigned long rootWin, parentWin;
   unsigned int numChildren;
-  unsigned long *childList;
+  unsigned long *childList = NULL;
 
   if (!XQueryTree(nxagentDisplay, window, &rootWin, &parentWin, &childList,
                       &numChildren))
```

Comment:
Hi Uli,

I think here is still a memleak when XQueryTree fails and childList is set.

Reply:
Ah okay I was wrong here. In the error case the malloc of the children has failed and we should not free it.